plug.vim

```diff
@@ -526,6 +531,14 @@ function! s:lod_map(map, names, with_prefix, prefix)
   call feedkeys(substitute(a:map, '^<Plug>', "\<Plug>", '') . extra)
 endfunction
 
+function! s:lod_autocmd(name)
+  call s:lod([a:name], ['ftdetect', 'after/ftdetect', 'plugin', 'after/plugin'])
+  call s:dobufread([a:name])
```

Comment:
This might be problematic.
I found this PR when looking for a better way to handle custom `InsertEnter` handling.

I am using the following:
```vim
    Plug 'Shougo/deoplete.nvim', {'on': []}
    …
    augroup vimplug_load_on_insertmode
      autocmd!
      autocmd InsertEnter * call plug#load(
            \ 'cursorcross.vim',
            \ 'delimitMate',
            \ 'deoplete.nvim',
            \ ) | call deoplete#enable()
            \ | autocmd! vimplug_load_on_insertmode
    augroup END
```

And here the call to `deoplete#enable()` adds a `BufEnter` autocmd that gets triggered then.
It is a bit weird, since `plug#load()` should be finished already, before `deoplete#enable()` then sets up the `BufEnter` autocmd?!  (I have also put it into a separate function to make sure `|` does not cause it).

Maybe the `BufEnter` gets queued after the `InsertEnter`?!